php/class-wp-https-detection.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * WP_HTTPS_Detection class.
+ *
+ * @package PWA
+ */
+
+/**
+ * WP_HTTPS_Detection class.
+ */
+class WP_HTTPS_Detection {
+
+	/**
+	 * The cron hook to check HTTPS support.
+	 *
+	 * @var string
+	 */
+	const CRON_HOOK = 'check_https_support';
+
+	/**
+	 * Option name for whether HTTPS is supported.
+	 *
+	 * @var string
+	 */
+	const HTTPS_SUPPORT_OPTION_NAME = 'is_https_supported';
+
+	/**
+	 * Secret for the HTTPS detection request.
+	 *
+	 * @var string
+	 */
+	const REQUEST_SECRET = 'https_detection_secret';
+
+	/**
+	 * Query arg key for the HTTPS detection token.
+	 *
+	 * @var string
+	 */
+	const REQUEST_TOKEN_QUERY_ARG = 'https_detection_token';
+
+	/**
+	 * Inits the class.
+	 */
+	public function init() {
+		add_action( 'wp', array( $this, 'schedule_cron' ) );
+		add_action( self::CRON_HOOK, array( $this, 'update_option_https_support' ) );
+		add_filter( 'cron_request', array( $this, 'ensure_http_if_sslverify' ), PHP_INT_MAX );
+	}
+
+	/**
+	 * Schedules a cron event to check for HTTPS support.
+	 */
+	public function schedule_cron() {
+		if ( ! wp_next_scheduled( self::CRON_HOOK ) ) {
+			wp_schedule_event( time(), 'hourly', self::CRON_HOOK );
+		}
+	}
+
+	/**
+	 * Makes a request to find whether HTTPS is supported, and stores the result in an option.
+	 */
+	public function update_option_https_support() {
+		update_option( self::HTTPS_SUPPORT_OPTION_NAME, $this->is_https_supported() );
+	}
+
+	/**
+	 * Makes a request to the home URL to determine whether HTTPS is supported.
+	 *
+	 * Returns true if the URL has the scheme of HTTPS, and it has the correct token.
+	 * The token is only to verify that the request is for the same site.
+	 * Also, in wp_remote_get(), 'sslverify' is true by default.
+	 *
+	 * @return boolean Whether HTTPS is supported.
+	 */
+	public function is_https_supported() {
```

Comment:
I think this would be marginally better-named `check_https_support()`. A `is_https_supported()` function I think would be more on the lines of doing `return (bool) get_option( self::HTTPS_SUPPORT_OPTION_NAME )`, in other words a direct replacement of `is_ssl()`.

Reply:
Thanks, [it's renamed](https://github.com/xwp/pwa-wp/pull/16/files#diff-1d91a59d72444acecd2a72e8baa63c2cR80) to `check_https_support()`.

Reply:
Also, here's the new method [is_https_supported()](https://github.com/xwp/pwa-wp/pull/16/files#diff-1d91a59d72444acecd2a72e8baa63c2cR56).